services/comprehension/feedback-api-main/src/endpoint/endpoint.go

```diff
@@ -171,6 +172,16 @@ func processResults(results map[int]InternalAPIResponse, length int) (int, bool)
 	return automl_index, all_correct
 }
 
+func safeSendToChannel(c chan InternalAPIResponse, response InternalAPIResponse) {
```

Comment:
So based on what I've read, we can't use this approach because the channel could close right after this `select` check, then you'd get a panic for sending to a closed channel. You can only close a channel if you are assured that you won't send anything else to it, e.g. a synchronous loop that adds things to a channel, then a close.

> Note: Only the sender should close a channel, never the receiver. Sending on a closed channel will cause a panic.
https://tour.golang.org/concurrency/4

I'm trying some other approaches.